plugins/inputs/nats/nats.go

```diff
@@ -0,0 +1,79 @@
+package nats
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"encoding/json"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	gnatsd "github.com/nats-io/gnatsd/server"
+)
+
+type Nats struct {
+	Server string
+}
+
+var sampleConfig = `
+  ## The address of the monitoring end-point of the NATS server
+  server = "http://localhost:1337"
+`
+
+func (n *Nats) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Nats) Description() string {
+	return "Provides metrics about the state of a NATS server"
+}
+
+func (n *Nats) Gather(acc telegraf.Accumulator) error {
+	theServer := fmt.Sprintf("%s/varz", n.Server)
+
+	/* download the page we are intereted in */
+	resp, err := http.Get(theServer)
```

Comment:
Can you add in a custom transport for this plugin?  The best example to follow is the [apache input](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/apache/apache.go#L100).  You don't have to add SSL support or user configurable timeouts, but make sure there is a timeout on the http.Client of 5 seconds.

Reply:
Done (exposed the timeout as an option too)